deploy-tool/dns/gcloud/gcloud.go

```diff
@@ -0,0 +1,169 @@
+package gcloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/libdns/libdns"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/dns/v1"
+	"google.golang.org/api/option"
+)
+
+// GoogleDNSProvider is a structure that defines the google DNS provider implementation
+type GoogleDNSProvider struct {
+	service      *dns.Service
+	project      string
+	zoneMappings map[string]string
+}
+
+const (
+	changeStatusDone = "done"
+)
+
+// NewGoogleDNSProvider creates a new GoogleDNSProvider with the given project name
+func NewGoogleDNSProvider(project string, zoneMappings map[string]string) (*GoogleDNSProvider, error) {
+	client, err := google.DefaultClient(context.Background(), dns.NdevClouddnsReadwriteScope)
+	if err != nil {
+		return nil, err
+	}
+
+	svc, err := dns.NewService(context.Background(), option.WithHTTPClient(client))
+	if err != nil {
+		return nil, err
+	}
+
+	return &GoogleDNSProvider{
+		service:      svc,
+		project:      project,
+		zoneMappings: zoneMappings,
+	}, nil
+}
+
+func convertRecordType(record libdns.Record) dns.ResourceRecordSet {
+	return dns.ResourceRecordSet{
+		Name:    fmt.Sprintf("%s.", record.Name),
+		Rrdatas: []string{record.Value},
+		Ttl:     record.TTL.Nanoseconds() / 1e9, // Expected in seconds
```

Comment:
nit: seems like you could use `time.Duration.Seconds()` and cast to the appropriate integer type.

Reply:
I didn't like how seconds was implemented on this: 
![image](https://user-images.githubusercontent.com/1465785/96943191-43c6a180-148c-11eb-8b29-a4d11e4ae127.png)

And it seemed like since everything is basing it off of nanoseconds then this one won't hurt either. I'm not sure if I'm comfortable letting the time be set by Go's time conversions since I don't know enough about their edge cases.